presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -237,13 +237,8 @@ private int readWithFilter(int[] positions, int positionCount)
                 boolean inRowDictionary = inDictionaryStream != null && !inDictionaryStream.nextBit();
                 int rawIndex = toIntExact(dataStream.next());
                 int index = inRowDictionary ? stripeDictionarySize + rawIndex : rawIndex;
-                int length;
-                if (isCharType) {
-                    length = maxCodePointCount;
-                }
-                else {
-                    length = inRowDictionary ? rowGroupDictionaryLength[rawIndex] : stripeDictionaryLength[rawIndex];
-                }
+                int length = dictionaryOffsetVector[index + 1] - dictionaryOffsetVector[index];
```

Comment:
@sdruzkin  this is the main change for this PR, What kind of testing is required to verify it ? Does unit test capture the changes